server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -295,13 +279,15 @@ private void allocateUnassigned() {
             ShardRouting[] secondary = new ShardRouting[primary.length];
```

Comment:
like maybe secondary here could be "deferred" or something, that's what seems to be happening?